modules/transactionpool/accept.go

```diff
@@ -198,24 +202,45 @@ func (tp *TransactionPool) handleConflicts(ts []types.Transaction, conflicts []T
 		tp.transactionListSize -= len(encoding.Marshal(conflictSet))
 		delete(tp.transactionSets, conflict)
 		delete(tp.transactionSetDiffs, conflict)
+		diff.RevertedTransactions = append(diff.RevertedTransactions, modules.TransactionSetID(conflict))
+		delete(tp.subscriberSets, conflict)
 	}
 
 	// Add the transaction set to the pool.
 	setID := TransactionSetID(crypto.HashObject(superset))
 	tp.transactionSets[setID] = superset
-	for _, diff := range cc.SiacoinOutputDiffs {
-		tp.knownObjects[ObjectID(diff.ID)] = setID
+	for _, txDiff := range cc.SiacoinOutputDiffs {
+		tp.knownObjects[ObjectID(txDiff.ID)] = setID
 	}
-	for _, diff := range cc.FileContractDiffs {
-		tp.knownObjects[ObjectID(diff.ID)] = setID
+	for _, txDiff := range cc.FileContractDiffs {
+		tp.knownObjects[ObjectID(txDiff.ID)] = setID
 	}
-	for _, diff := range cc.SiafundOutputDiffs {
-		tp.knownObjects[ObjectID(diff.ID)] = setID
+	for _, txDiff := range cc.SiafundOutputDiffs {
+		tp.knownObjects[ObjectID(txDiff.ID)] = setID
 	}
```

Comment:
@DavidVorick are we sure that we are not missing ids of the deduplicated set here by not using `relatedObjectIDs` ?